factorio-map-process.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+## Defaults
+SERVER_BASE_PATH=/srv/factorio/maps
+MAP_TILES_PATH=images
+FACTORIO_MAP_SCRIPT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"/factoriomap.py
+TARFILE=
+REDUCE=0
+
+function usage() {
+    cat <<-EOF
+USAGE
+
+factorio-map-process.sh [OPTIONS] tarfile
+
+DESCRIPTION
+
+This script will process a given tarfile into a set of images used in a leaflet map.
+The default behavior will create a directory for the world maps if one does not exist and copy the example html page in as 'index.html'
+The tar file will be parsed for a map name and a date, separated by underscores (_). Files should be of the format {WorldName}_YYYY_MM_DD.tar.
+    
+OPTIONS
+
+    -t | --tiles <images>           Set the name of the directory to store the tile images in.  Default value is
+                                    'images'
+
+    -s | --server <server path>     Set the base path where the http server files will be stored.  Default value is
+                                    '/srv/factorio/maps'
+
+    -r                              Flag to size-reduce the images folder with the rdfind tool
+EOF
+}
+
+function testRequirements() {
+	if ! [ -x "$(command -v jq)" ]; then
+		echo "ERROR: Requirement not satisfied - jq -"
+		exit 1
+	fi
+
+	if ! [ -x "$(command -v python3)" ]; then
+		echo "ERROR: Requirement not satisfied - python3 -"
```

Comment:
This test seems a little pointless, since you are testing for `python3`, but not the other dependencies of `factoriomap.py`.

Reply:
I think testing for python3 is fair, it's a top-level check and doesn't assume much about the user's environment.  Also, the dependencies of the python script should be tested by the python script, otherwise updates to the python would need to be also updated here.